### ISSUE TITLE
Graphics Don't Output to PDF

### ISSUE THREAD

**Issue by acuffza** [RhoInc/sas-codebook#15]
I'm running the macro "codebook_generic" and the PDF looks great, but the graphic for each variable doesn't end up on the PDF. Only the graphic for the first variable appears. The graphics just get output to PNG files in my data folder. I've only specified the macro variables "DATA" and "PDFPREFIX". I'm using SAS/STAT 14.1. Do you know what might be going on?

**Comment by srosanba** [comment]
acuffza,
Off the top, I'm not sure what's going on. I've seen weird things, but not that particular weird thing. A couple of thoughts:
1. Near as I can tell SAS/STAT 14.1 means you are using BASE version 9.4M3. Can you confirm that you are on 9.4M3 (automatic macro variable SYSVLONG)?
2. Is your data confidential? If not, could you share your program and data? If so, could you try to replicate on non-confidential data and share that program and data?

**Comment by acuffza** [comment]
Thanks so much for your response! The SYSVLONG macro variable resolves to "9.04.01M3P062415", so it looks like I'm using the correct version. The data is confidential, but I get the same problem on any dataset so I ran the macro for the SAS dataset "CARS". I attached a text file of the exact SAS code I used and the PDF that I got from it. Thank you again!

[Try Codebook Text.txt](https://github.com/RhoInc/sas-codebook/files/271380/Try.Codebook.Text.txt)
[sashelp.cars.pdf](https://github.com/RhoInc/sas-codebook/files/271382/sashelp.cars.pdf)

**Comment by srosanba** [comment]
It took a while, but I was finally able to gain access to a development instance of 9.4M3 (we use 9.4M2 in production). My results match yours: the images are not being included as part of the output when I run it in M3. I will work on this and see what I can figure out. I'll post again once I have something for you.

**Comment by acuffza** [comment]
Thank you so much, I really appreciate it!

**Comment by acuffza** [comment]
I contacted SAS about the problem and they suggested the following change:

Change this title:

```
  title1 
     j=l "~S={cellwidth=6.5in}Abbr. codebook for %upcase(&&data&d) (&varlist_n vars, &&nobs&d obs)" 
     j=r 'Page ~{thispage} of ~{lastpage}'
     ;
```

To:

```
  title1 
     j=l "~S={cellwidth=6.5in}Abbr. codebook for %upcase(&&data&d) (&varlist_n vars, &&nobs&d obs)" 
     ; *j=r 'Page ~{thispage} of ~{lastpage}'         ;
```

I got the graphics to output properly this way! He said he suspected it had something to do with this issue: http://support.sas.com/kb/34/573.html

Thanks again for sharing this code, it has proven incredibly useful!

**Comment by srosanba** [comment]
Thanks for working with SAS to troubleshoot. It was on my list, but you beat me to it. I like having the ~{lastpage} in the report, and it works in 9.4m2 (which is what we have at my organization), but I think for the greater good it is probably best if I just remove the ~{lastpage} from the code.
Thanks again for figuring this out. Much appreciated!

**Comment by srosanba** [comment]
acuffza,
two question: 
1. how did you stumble upon the codebook macro? 
2. do you have any comments/criticisms/suggestions?
thanks!

**Comment by acuffza** [comment]
1. My supervisor wanted me to find a program that automatically generated a codebook for a SAS dataset, so I did some Googling. After exploring a few of the options I found, your program looked like it gave the best-looking results, had the best comments/descriptions, and was the most user-friendly.
2. My only suggestion would be that it would be especially nice if it were possible to break up all of the variable summary statistics by the values of some other variable. If there were a treatment/control variable, for example, it would be nice to see all the variable summaries for all the "Treatment" observations and then have the "Control" variable summaries appear next to that. However, I realize that that would require a great deal of reworking and isn't really the point of the program. As far as doing what the program sets out to accomplish (provide a nice snapshot of each variable in a dataset), I don't think it could do a better job than it already does.
